src/domain/use-cases/pet/breed/load-dog-breeds.ts

```diff
@@ -0,0 +1,13 @@
+import { type LoadDogBreedsRepository } from '@/data/protocols/db/breed'
+import { type Breed } from '@/domain/models/breed'
+
+export interface LoadDogBreeds {
+  load: () => Promise<LoadDogBreeds.Result>
+}
+
+export namespace LoadDogBreeds {
+  export type Result = Breed[] | undefined
```

Comment:
i think that don't need `| undefined`, because if has no breeds just return an empty list

Reply:
the typescript needs that because it is getting something of the DB, soo, maybe something happen wrong, because of that its necessary put undefined there.